src/orquestra/sdk/_base/_api/_wf_run.py

```diff
@@ -875,3 +878,64 @@ def _parse_max_age(age: t.Optional[str]) -> t.Optional[timedelta]:
         '- "10m" = 10 minutes,\n'
         '- "3D6h8M13s" = 3 days, 6 hours, 8 minutes and 13 seconds.'
     )
+
+
+def _get_workspace_and_project_ids() -> (
+    t.Tuple[t.Optional[WorkspaceId], t.Optional[ProjectId]]
+):
+    if not os.getenv(_env.CURRENT_CLUSTER_ENV):
+        return None, None
+
+    return os.getenv(_env.CURRENT_WORKSPACE_ENV), os.getenv(_env.CURRENT_PROJECT_ENV)
+
+
+def _generate_cluster_uri_name(uri: str) -> str:
+    return str(urlparse(uri).path).split(".")[0]
+
+
+def _get_config_context() -> str:
+    cluster_uri = os.getenv(_env.CURRENT_CLUSTER_ENV)
+    if not cluster_uri:
+        context = _exec_ctx.get_current_exec_context()
+        if context == _exec_ctx.ExecContext.RAY:
+            return _config.RAY_CONFIG_NAME_ALIAS
+        elif context == _exec_ctx.ExecContext.DIRECT:
+            return _config.IN_PROCESS_CONFIG_NAME
+        else:
+            raise NotImplementedError(
+                f"Got unexpected global context {context}. Please report this as a bug."
+            )
+
+    return _generate_cluster_uri_name(cluster_uri)
+
+
+class CurrentWorkflowIDs(t.NamedTuple):
+    config_name: ConfigName
+    workspace_id: t.Optional[WorkspaceId]
+    project_id: t.Optional[ProjectId]
+
+
+def current_wf_ids() -> CurrentWorkflowIDs:
+    """Get the backend IDs related to current workflow execution context.
```

Comment:
**thought**: this oneliner is good; this is where I've got the idea for alternative naming from